OpenOversight/app/utils.py

```diff
@@ -77,6 +77,16 @@ def dept_choices():
     return db.session.query(Department).all()
 
 
+def year_choices():
+    oldest_year = db.session.query(cast(func.extract('year', Officer.employment_date), Integer)) \
+                            .filter(Officer.employment_date.isnot(None)) \
+                            .order_by(func.extract('year', Officer.employment_date)) \
+                            .limit(1).first()[0]
+    current_year = datetime.datetime.now().year
+    year_choices = [str(year) for year in range(oldest_year, current_year + 1)]
+    return year_choices
```

Comment:
This looks good. I think for the oldest year we should use a `SELECT min(employment_date) FROM officers` to make it as efficient as possible. I looked up how to do that in sqlalchemy and it seems
```oldest_year = db.session.query(func.min(Officer.employment_date)).scalar().year```
should be do the trick.
What do we want to do in the case that there is no officer with a (not NULL) employment date?

Reply:
I keep coming back to this and I am think we should just hard-code the `oldest_year`. I think we now have some officers in the database whose employment_date is before 1950, which will make the drop-down menu pretty long and we really don't have good data for those year anyways. So my suggestion would be to hard-code it to 1980 or something, but I am also fine with the suggestion I made, if we can come up with a solution for an officer table without any non-NULL employment dates.